### PR TITLE
Fix for issue #1993

### PR DESCRIPTION
**Pull request by prinzdezibel**
Resolves: #1993 
Safari incognito mode does not support local storage.

 - This prevents re-login through local storage
 - Can happen whenever app is navigating away from home domain, e.g. Paypal-Express checkout flow
 
The fix is to have the application set a login token as cookie whenever the page is going to be unloaded (window.onbeforeunload). After the external process the app will be initialized again: A good place to read the cookie again and log in.

### CODE REVIEW

**Comment by prinzdezibel** [comment]
@aaronjudd  Sorry for being a bit late with this PR, even though I offered it quite a while ago. I was very busy the last couple of months.

File: client/modules/core/startup.js [hunk 1]
```diff
@@ -2,7 +2,12 @@ import store from "amplify-store";
 import { Meteor } from "meteor/meteor";
 import { Tracker } from "meteor/tracker";
 import { Accounts } from "meteor/accounts-base";
+
 import Reaction from "./main";
+import Logger from "/client/modules/logger";
```
Comment (zenweasel) [inline client/modules/core/startup.js]: I think we want to import `Reaction` and `Logger` from `.client/api` like we do elsewhere for consistency.

File: client/modules/core/startup.js [hunk 2]
```diff
@@ -14,17 +19,66 @@ Meteor.startup(function () {
   // initialize anonymous guest users
   return Tracker.autorun(function () {
     const userId = Meteor.userId();
+
+    if (userId && !isLocalStorageAvailable() && !readCookie(cookieName)) {
+      Logger.info("No local storage available. About to set up fall-back login " +
```
Comment (zenweasel) [inline client/modules/core/startup.js]: This should probably not be an `info` level log.

File: client/modules/core/startup.js (hunk 1 above)
Comment (prinzdezibel) [inline client/modules/core/startup.js]: sure. I'll fix.

File: client/modules/core/startup.js (hunk 2 above)
Comment (prinzdezibel) [inline client/modules/core/startup.js]: what log level do you find appropriate?

Comment (zenweasel) [inline client/modules/core/startup.js]: debug?

Comment (zenweasel) [inline client/modules/core/startup.js]: we're all about the `debug` these days. `info` is so 2016

Comment (prinzdezibel) [inline client/modules/core/startup.js]: :)

File: client/modules/core/startup.js (hunk 1 above)
Comment (prinzdezibel) [inline client/modules/core/startup.js]: done.

File: client/modules/core/startup.js (hunk 2 above)
Comment (prinzdezibel) [inline client/modules/core/startup.js]: done.

**Pull request merged**